File: GeneratedEntityReaderClasses/Classes/PixelSprite.cs
namespace GeneratedNoitaClasses;
public class PixelSprite :  iNoitaType<PixelSprite>{
	public NoitaString Number1 { get; set; } = new();
	public NoitaFloat Number2 { get; set; } = new();
	public NoitaFloat Number3 { get; set; } = new();
	public NoitaFloat Number4 { get; set; } = new();
	public NoitaFloat Number5 { get; set; } = new();
	public NoitaInt Number6 { get; set; } = new();
	public NoitaInt Number7 { get; set; } = new();
	public NoitaInt Number8 { get; set; } = new();
	public NoitaBool Number9 { get; set; } = new();
	public NoitaInt Number10 { get; set; } = new();
	public NoitaString Number11 { get; set; } = new();
	public NoitaSpecialTexture Number12 { get; set; } = new();

    public void Read(NoitaStream s){
		Number1.Read(s);
		Number2.Read(s);
		Number3.Read(s);
		Number4.Read(s);
		Number5.Read(s);
		Number6.Read(s);
		Number7.Read(s);
		Number8.Read(s);
		Number9.Read(s);
		Number10.Read(s);
		Number11.Read(s);
		Number12.Read(s);

    }
    public void Write(NoitaStream s){
		Number1.Write(s);
		Number2.Write(s);
		Number3.Write(s);
		Number4.Write(s);
		Number5.Write(s);
		Number6.Write(s);
		Number7.Write(s);
		Number8.Write(s);
		Number9.Write(s);
		Number10.Write(s);
		Number11.Write(s);
		Number12.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/PixelSprite.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaFloat
from BasicNoitaTypes import NoitaInt
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaSpecialTexture

from NoitaStream import NoitaStream
@dataclass
class PixelSprite:
    Number1:NoitaString = field(default_factory=NoitaString)
    Number2:NoitaFloat = field(default_factory=NoitaFloat)
    Number3:NoitaFloat = field(default_factory=NoitaFloat)
    Number4:NoitaFloat = field(default_factory=NoitaFloat)
    Number5:NoitaFloat = field(default_factory=NoitaFloat)
    Number6:NoitaInt = field(default_factory=NoitaInt)
    Number7:NoitaInt = field(default_factory=NoitaInt)
    Number8:NoitaInt = field(default_factory=NoitaInt)
    Number9:NoitaBool = field(default_factory=NoitaBool)
    Number10:NoitaInt = field(default_factory=NoitaInt)
    Number11:NoitaString = field(default_factory=NoitaString)
    Number12:NoitaSpecialTexture = field(default_factory=NoitaSpecialTexture)
    
    def Read(self, s:NoitaStream):
        self.Number1.Read(s)
        self.Number2.Read(s)
        self.Number3.Read(s)
        self.Number4.Read(s)
        self.Number5.Read(s)
        self.Number6.Read(s)
        self.Number7.Read(s)
        self.Number8.Read(s)
        self.Number9.Read(s)
        self.Number10.Read(s)
        self.Number11.Read(s)
        self.Number12.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Number1.Write(s)
        self.Number2.Write(s)
        self.Number3.Write(s)
        self.Number4.Write(s)
        self.Number5.Write(s)
        self.Number6.Write(s)
        self.Number7.Write(s)
        self.Number8.Write(s)
        self.Number9.Write(s)
        self.Number10.Write(s)
        self.Number11.Write(s)
        self.Number12.Write(s)

